app/utils/UrlHelper.scala

```diff
@@ -46,11 +49,19 @@ class UrlHelper @Inject()(val appConfig: FrontendAppConfig) {
     appConfig.fulfilmentHouse
   }
 
-  def getPortalURL(key: String) = {
-    appConfig.getPortalUrl(key)
-  }
-
   def getHmceURL(key: String) = {
     appConfig.getHmceURL(key)
   }
+
+  def getPortalURL(key: String)(implicit request: Request[_]): String = {
+    val portalUrl = appConfig.getPortalUrl(key)
+    appendLanguage(portalUrl)(request)
+  }
+
+  private def appendLanguage(url: String)(implicit request: Request[_]) = {
+    val lang = if (LanguageUtils.getCurrentLang == LanguageUtils.Welsh) "lang=cym" else "lang=eng"
+    val token = if (url.contains("?")) "&" else "?"
```

Comment:
What if the url ends with a `?`, this would remove lang and create a nameless parameter

Reply:
In our case - URL's are written by us currently not generated, so I am not hoping this to happen. we can sit together to try this by writing test.